src/wallet/spend.cpp

```diff
@@ -1162,7 +1162,13 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
             // If an input has a preset sequence, we can't do anti-fee-sniping
             use_anti_fee_sniping = false;
         }
-        txNew.vin.emplace_back(coin->outpoint, CScript(), sequence.value_or(default_sequence));
+        CScript script_sig;
+        CScriptWitness script_witness;
+        if (coin_control.HasScripts(coin->outpoint)) {
+            std::tie(script_sig, script_witness) = coin_control.GetScripts(coin->outpoint);
+        }
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/25273/commits/2a06c4c2b33c614cc3f877bdc013c8df9e847ca9:

could be something like:

```suggestion
        auto scripts = GetScripts(coin->outpoint);
        if (scripts) {
            std::tie(script_sig, script_witness) = *scripts;
        }
````

if you decide to drop the "Has" method.